src/config_design.config

```diff
@@ -0,0 +1,126 @@
+
+   {put, [{count,1},  % Alternate count in more complex format, modify existing lookups
+          {key_generator, {int_to_str, {partitioned_sequential_int, 100000}}},
+          {value_generator, {function,http_couchdb_driver,make_json_doc,[]}}
+	 ]}
+   ]}.
+
+%% IMPLEMENTATION:
+%% basho_bench_worker:init 
+%%    turns per-op config into new Keygen/ValueGen entities
+%%    and stores into worker State as per-op-config list
+%%        per_op_config =[{put, [{keygen, KeyGen}, {valgen, ValGen}, ...]}, ... ]
+%% basho_bench_worker:worker_next_op2 
+%%    checks per_op_config, uses as run call arguments,
+%%       potentially stores onto State as current keygen/valgen
+
+%% ----------------------------------------------------------------------------
+%% WORKER_GROUP SUPPORT: https://github.com/cloudant/perf/issues/32
+%% ----------------------------------------------------------------------------
+%%
+%% "worker_types" would provide per-worker config list,
+%%     miniature copy of current config file for what is different from global
+
+{worker_types, [
+   %% Each worker_type can have its own list of config options
+   %%     to be used on top of global parameters known to basho_bench_config
+   {worker_name_a, [ {weight, 5}, 
+                     {driver, cb_driver}, 
+                     {operations, [
+                         {get,1},
+                         {put, [ {count,1}, 
+                                 {key_generator, keyGen1}
+                               ]}
+                         ]}, 
+                     {other, 1}
+                   ]},
+   {worker_name_b, []}
+   ]}.
+
+{workers, [
+   {worker_name_a, 1},
+   {worker_name_b, 3}
+   ]}.
+
+%% IMPLEMENTATION:
+%% The "concurrent" property picks how many workers but use workers weights for proportions needed. 
+%%
+%% basho_bench_worker_config:next_worker 
+%%    pops worker_name off weighted list based on "workers" proportions
+%%
+%% basho_bench_worker:init 
+%%    if next_worker is defined then 
+%%        looks up worker type to get per-worker-type config as LocalConfig
+%%        uses config_get(Key, LocalConfig) for initialization
+%%        stores LocalConfig into state for future use
+%%        benchmark applications can use config_get and then save values into own driverState
+%% 
+%% basho_bench_worker:config_get(Key,Default,LocalConfig) 
+%%    checks LocalConfig first then uses basho_bench_config:get for global defaults
+%%
+%% ISSUES
+%% 1. Some parms (source_dir, pre/post_hook, etc) and funs like log_dimensions may need more thought
+%%    to handle worker_types since these are currently handled outside basho_bench_worker.
+%%    May end up with list of pre- and post-hooks to run across multiple drivers, etc. 
+
+
+%% ----------------------------------------------------------------------------
+%% SETUP/TEARDOWN: https://github.com/cloudant/perf/issues/35
+%% ----------------------------------------------------------------------------
+%%
```

Comment:
What do we need a "setup" function to do that the existing driver "init" function isn't already doing ? 
If we just document the "terminate" function would that cover the "teardown" requirement ? 

We also have pre- and post-hooks that are run before workers are created and after they've all been shutdown. For worker groups we need to be able to run these hooks per-driver (i.e. MULTIPLE hooks in a given run) but with per-driver (hooks) and per-worker (init/terminate) we may have the basics already in place.